### PR TITLE
Add buckets on lower end of distribution to network bridge latency metrics

### PR DESCRIPTION
**Pull request by bredamatt**
Similar to a previous PR. Adds a couple of more buckets to the network bridge latency metrics. As noticeable below, all observations are currently below 50ms, and although trivial, this PR adds two more buckets for the lower end of the distribution which could be somewhat useful for future network improvements.

Screenshot 2022-11-28 at 15 01 54

### CODE REVIEW

**Comment by sandreim** [comment]
I would imagine most of the samples are < 5ms. Did we try this in Zombienet to see what are the actual values that we want to study their distribution ?

**Comment by vstakhov** [comment]
I'd suggest to remove some high latency buckets to compensate additional more precise low end buckets. It also makes perfect sense counting that the graphs clearly demonstrate that the high end buckets are not populated.

**Comment by bredamatt** [comment]
> I would imagine most of the samples are < 5ms. Did we try this in Zombienet to see what are the actual values that we want to study their distribution ?

Just tried this now, but these results are quite misleading and not representative as real network values. It will always be a lot faster on one host (its IPC effectively). Also, it appears that every bucket is incremented per observation, whilst the sum is very very small, so it is hard to say how many observations actually fall within each bucket: 

```
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.05"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.1"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.2"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.3"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.4"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.5"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.6"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.75"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.9"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="1"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="1.2"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="1.5"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="1.75"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="+Inf"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="circulate_statement",chain="rococo_local_testnet"} 0.00000045100000000000005
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="circulate_statement",chain="rococo_local_testnet"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.05"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.1"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.2"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.3"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.4"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.5"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.6"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.75"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.9"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="1"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="1.2"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="1.5"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="1.75"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="+Inf"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="handle_incoming_message",chain="rococo_local_testnet"} 0.000000617
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="handle_incoming_message",chain="rococo_local_testnet"} 25
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.05"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.1"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.2"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.3"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.4"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.5"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.6"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.75"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.9"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="1"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="1.2"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="1.5"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="1.75"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="+Inf"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="new_gossip_topology",chain="rococo_local_testnet"} 0.00000020500000000000002
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="new_gossip_topology",chain="rococo_local_testnet"} 10
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.05"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.1"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.2"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.3"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.4"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.5"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.6"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.75"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.9"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="1"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="1.2"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="1.5"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="1.75"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="+Inf"} 143
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="peer_view_change",chain="rococo_local_testnet"} 0.0000034530000000000023
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="peer_view_change",chain="rococo_local_testnet"} 143
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2096690

**Comment by sandreim** [comment]
> > I would imagine most of the samples are < 5ms. Did we try this in Zombienet to see what are the actual values that we want to study their distribution ?
> 
> @sandreim Just tried this now, but these results are quite misleading and not representative as real network values. It will always be a lot faster on one host. Also, it appears that every bucket is incremented per observation, whilst the sum is very very small, so it is hard to say how many observations actually fall within each bucket just by looking at the output like this. Am I right in thinking that whenever bucket with `le=0.05` is incremented, then all buckets with `le > 0.05` are also incremented? Nevertheless, it does not really answer how low the values go I think.

All samples are in 0.05 in this output. Can you try this with your new buckets ? The idea was to calibrate the buckets based on the observations

**Comment by sandreim** [comment]
you can do `sum/count` to get the average value which is helpful

**Comment by bredamatt** [comment]
As predicted by @sandreim even lower buckets don't really improve this visualisation. Here is the output from even smaller buckets than what is introduced by this PR (0.0025, 0.005, 0.1, 0.025, 0.05):

```
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.0025"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.005"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.01"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.025"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.05"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.1"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.2"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.3"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.4"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.5"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.6"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="0.75"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="circulate_statement",chain="rococo_local_testnet",le="+Inf"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="circulate_statement",chain="rococo_local_testnet"} 0
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="circulate_statement",chain="rococo_local_testnet"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.0025"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.005"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.01"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.025"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.05"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.1"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.2"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.3"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.4"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.5"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.6"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="0.75"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="handle_incoming_message",chain="rococo_local_testnet",le="+Inf"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="handle_incoming_message",chain="rococo_local_testnet"} 0
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="handle_incoming_message",chain="rococo_local_testnet"} 1
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.0025"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.005"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.01"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.025"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.05"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.1"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.2"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.3"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.4"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.5"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.6"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="0.75"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="new_gossip_topology",chain="rococo_local_testnet",le="+Inf"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="new_gossip_topology",chain="rococo_local_testnet"} 0.000000041
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="new_gossip_topology",chain="rococo_local_testnet"} 3
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.0025"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.005"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.01"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.025"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.05"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.1"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.2"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.3"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.4"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.5"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.6"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="0.75"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_bucket{message_type="peer_view_change",chain="rococo_local_testnet",le="+Inf"} 36
polkadot_parachain_statement_distribution_network_bridge_update_v1_sum{message_type="peer_view_change",chain="rococo_local_testnet"} 0.0000008210000000000002
polkadot_parachain_statement_distribution_network_bridge_update_v1_count{message_type="peer_view_change",chain="rococo_local_testnet"} 36
```

File: node/network/statement-distribution/src/metrics.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ use polkadot_node_subsystem_util::metrics::{self, prometheus};
 
 /// Buckets more suitable for checking the typical latency values
 const HISTOGRAM_LATENCY_BUCKETS: &[f64] =
-	&[0.01, 0.025, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.75, 0.9, 1.0, 1.2, 1.5, 1.75];
+	&[0.000025, 0.00005, 0.000075, 0.0001, 0.0003125, 0.000625, 0.00125, 0.0025, 0.005, 0.01, 0.025, 0.05, 0.1];
```
Comment (bredamatt) [inline node/network/statement-distribution/src/metrics.rs:21]: Locally these buckets don't make a big difference, but I am interested in seeing if they do in a more realistic scenario.

Comment (sandreim) [reply node/network/statement-distribution/src/metrics.rs:21]: For some realistic values I would look at some historical data from Versi. Even if we don't have these buckets, the sum and count of samples allow us to compute some average value which could be used to estimated some good buckets.

Comment (bredamatt) [reply node/network/statement-distribution/src/metrics.rs:21]: Good idea!

**Review by vstakhov — APPROVED** [review]
It looks matching with the data we have (I'm not 100% sure about 25 usec bucket though). It would also be nice to fix the `fmt` failure.

**Review by m-cat — APPROVED** [review]

**Review by tdimitrov — APPROVED** [review]

**Comment by sandreim** [comment]
bot merge

**Pull request merged**